doc/contributor/launchers.md

```diff
@@ -0,0 +1,38 @@
+# Executables in `bin` 
+
+## Requirements
+
+1.  [x] When TruffleRuby tar (GraalVM) is downloaded and unpacked, all files in `bin` and `language/ruby/bin` has to 
+    always run the TruffleRuby from the tar. 
+    -   Therefore `PATH` cannot be involved (`/usr/bin/env` cannot be used in shebang).
+    -   There cannot be absolute paths in `bin` executables.
+    -   All scripts in `bin` have to always resolve to `truffleruby` in the same dir.
+2.  [x] Newly installed gem executables has to always run the `truffleruby` from the same `bin`. 
+    -   By default RubyGems generate executables with absolute path.
+3.  [x] Behave as proper Ruby implementation when the `bin` is put on `PATH`.
+4.  [ ] RubyMine requires `gem` in `bin` to be Ruby (otherwise gems are not listed).
+    -   `gem` file in `bin` has to be Ruby or the `-x` option used by Rubymine has to be extended to understand our bash files in `bin`
```

Comment:
The relevant logic is in https://github.com/ruby/ruby/blob/581c447ff93bdd8576e8532bc68891659a78d44e/ruby.c#L1779-L1807
Line 1806 decides to look for a ruby shebang if it finds a shebang not corresponding to ruby_engine on the first line of the script.